reconcile/dashdotdb_dvo.py

```diff
@@ -0,0 +1,105 @@
+import logging
+import os
+
+import requests
+
+from reconcile import queries
+from utils import threaded
+from utils.oc import OC_Map
+from utils.oc import StatusCodeError
+from utils.secret_reader import SecretReader
+
+LOG = logging.getLogger(__name__)
+QONTRACT_INTEGRATION = 'dashdotdb-dvo'
+DASHDOTDB_SECRET = os.environ.get('DASHDOTDB_SECRET',
+                                  'app-sre/dashdot/auth-proxy-production')
+
+
+class DashdotdbDVO:
+    def __init__(self, dry_run, thread_pool_size):
+        self.dry_run = dry_run
+        self.thread_pool_size = thread_pool_size
+        self.settings = queries.get_app_interface_settings()
+        secret_reader = SecretReader(settings=self.settings)
+        secret_content = secret_reader.read_all({'path': DASHDOTDB_SECRET})
+        self.dashdotdb_url = secret_content['url']
+        self.dashdotdb_user = secret_content['username']
+        self.dashdotdb_pass = secret_content['password']
+        self.prom_user = secret_content['prom_username']
+        self.prom_pass = secret_content['prom_password']
+
+    def _post(self, deploymentvalidation):
+        if deploymentvalidation is None:
+            return None
+
+        cluster = deploymentvalidation['cluster']
+        dvdata = deploymentvalidation['data']
+        response = None
+        if not self.dry_run:
+            endpoint = (f'{self.dashdotdb_url}/api/v1/'
+                        f'deploymentvalidation/{cluster}')
+            response = requests.post(url=endpoint, json=dvdata,
+                                     auth=(self.dashdotdb_user,
+                                           self.dashdotdb_pass))
+            try:
+                response.raise_for_status()
+            except requests.exceptions.HTTPError as details:
+                LOG.error('DVO: error posting %s - %s', cluster, details)
+
+        LOG.info('DV: cluster %s synced', cluster)
+        return response
+
+    def promql(url, query, auth=None):
```

Comment:
Does it make sense to use https://pypi.org/project/prometheus-api-client/?

Reply:
I started out using this but had some issues with how long things were taking / timeouts, unfortunately found these were unrelated to the prom lib but also all I'm doing with prom is fetching a raw json blob vs doing any prom object operations so at this time don't need the entire library anyway.

Reply:
replaced with self._promget()

Reply:
fair enough